### PR TITLE
Refine wasm/aot function instance lookup

### PR DESCRIPTION
**Pull request by wenyongh**
Sort the module instance's export functions with the function name,
and use binary search to lookup the wasm function.

### CODE REVIEW

**Review by lum1n0us — APPROVED** [review]
LGTM. 

followings include giving name to `wasm_exter_t` in `interp_process_export()` and `aot_process_export()`.

**Review by xujuntwt95329 — APPROVED** [review]
LGTM

File: core/iwasm/common/wasm_c_api.c [hunk 1]
```diff
@@ -3390,9 +3390,8 @@ wasm_func_call(const wasm_func_t *func, const wasm_val_vec_t *params,
                 AOTExport *export = module_aot->exports + export_i;
                 if (export->kind == EXPORT_KIND_FUNC) {
                     if (export->index == func->func_idx_rt) {
-                        func_comm_rt =
-                            (AOTFunctionInstance *)inst_aot->export_functions
-                            + export_func_j;
+                        func_comm_rt = aot_lookup_function(
+                            (AOTModuleInstance *)inst_aot, export->name);
```
Comment (yamt) [inline core/iwasm/common/wasm_c_api.c:3394]: is this cast necessary?

Comment (wenyongh) [reply core/iwasm/common/wasm_c_api.c]: yes, no need to cast again. fixed, thanks.

**Pull request merged**